telethon/client/users.py

```diff
@@ -51,6 +51,7 @@ async def _call(self: 'TelegramClient', sender, request, ordered=False):
 
         request_index = 0
         self._last_request = time.time()
+        attempt = 0
         for attempt in retry_range(self._request_retries):
```

Comment:
Seems pretty like an anti-pattern for me, though I see why you've added `attempt` assignment there.
Can it be rewritten without it? If no, then at least add a whitespace line before assignment.